STYLE_GUIDE.md

```diff
@@ -0,0 +1,443 @@
+  // ...
+  ```
+
+- Blank lines between __one-line__ functions or primitives may be omitted if they are related.
+
+  ```pony
+  primitive Red    fun apply(): U32 => 0xFF0000FF
+  primitive Green  fun apply(): U32 => 0x00FF00FF
+  primitive Blue   fun apply(): U32 => 0x0000FFFF
+
+  type Color is (Red | Green | Blue)
+  ```
+
+### Whitespace
+
+- Use spaces around binary operators and after commas, colons, and semicolons. Do not put spaces around brackets or parentheses.
+
+  ```pony
+  let x: USize = 3
+
+  x = x + (5 * x)
+
+  (let a, let b) = ("a", "b")
+
+  let ns = [as USize: 1; 2; 3]
+  ```
+
+- The `not` operator is surrounded by spaces, but the `-` operator is not followed by a space.
+
+  ```pony
+  if not x then -a end
+  ```
+
+- Lambda expressions follow the rules above except that a space is placed before the closing brace.
+
+  ```pony
+  // OK
+  {(a: USize, b: USize): USize => a + b }
+
+  // Not OK
+  {(a: USize, b: USize): USize => a + b}
+
+  // Not OK
+  { (a: USize, b: USize): USize => a + b }
+  ```
+
+- Normal rules apply to Lambda types. 
+
+  ```pony
+  // OK
+  type foo is {(USize, USize): USize} box
+
+  // Not OK
+  type foo is {(USize, USize): USize } box
+
+  // Not OK
+  type foo is { (USize, USize): USize } box
+  ```
+
+### Type Aliases
+
+Align long union and intersection types with each type on a separate line starting with the `(`, `|`, or `&` symbols. In this case, every symbol must be separated by a single space or a newline.
+
+```pony
+// OK
+type Signed is (I8 | I16 | I32 | I64 | I128)
+
+// OK
+type Signed is
+  ( I8
+  | I16
+  | I32
+  | I64
+  | I128 )
+
+// OK
+type Signed is
+  ( I8
+  | I16
+  | I32
+  | I64
+  | I128
+  )
+
+// Not OK
+type Signed is ( I8
+              | I16
+              | I32
+              | I64
+              | I128 )
+
+// Not OK
+type Signed is
+  (I8
+  |I16
+  |I32
+  |I64
+  |I128)
+```
+
+### Array Literals
+
+Multiline arrays generally have a newline between each entry, though logical exceptions exist (such as tables).
+
+```pony
+// OK
+let ns = [as USize: 1; 2; 3]
+
+// OK
+let strs = ["a"; "b"; "c"]
+
+// OK
+let ns =
+  [ as USize:
+    1
+    2
+    3 ]
+
+// OK
+let ns =
+  [ as USize:
+    1
+    2
+    3
+  ]
+
+// Not OK
+let ns =
+  [as USize:
+   1
+   2
+   3
+  ]
+
+// Not OK
+let ns = [ as USize:
+    1
+    2
+    3
+]
+
+// Not OK
+let ns = [ as USize:
+          1
+          2
+          3 ]
+```
+
+### Control Structures
+
+- Control structures are aligned so that their keywords (if, else, for, try, end, etc.) are at the same indentation level.
+
+  ```pony
+  if cond1 then e1
+  elif cond2 then e2
+  else e3
+  end
+
+  if cond1 then
+    e1
+  elif cond2 then
+    e2
+  else
+    e3
+  end
+
+  for name in ["Bob"; "Fred"; "Sarah"].values() do
+    env.out.print(name)
+  end
+  ```
+
+- Small expressions may be placed on a single line.
+
+  ```pony
+  if cond then e1 else e2 end
+  ```
+
+- Match expression cases are __not__ indented.
```

Comment:
This can be a bit misleading - it might be read as implying that match expression cases with multiple lines of body should not be indented, and I think we agree that they absolutely should.

We probably need an example showing this.